### ISSUE TITLE
initalizer-job does't seem to be idempotent causing some exceptions upon consecutive initializations

### ISSUE THREAD

**Issue by ccojocar** [DefectDojo/django-DefectDojo#2680]
**Bug description**

The `initializer-job` part of the helm chart which runs all migrations before the deployment, it doesn't seem to be idempotent when executed on an already initialized DB. The following exceptions are thrown:
```
enabling audit logging
Waiting for database to be reachable ..
enabling audit logging
Popen(['git', 'version'], cwd=/app, universal_newlines=False, shell=None, istream=None)
Popen(['git', 'version'], cwd=/app, universal_newlines=False, shell=None, istream=None)
patching TagDescriptor
No changes detected in app 'dojo'
enabling audit logging
Popen(['git', 'version'], cwd=/app, universal_newlines=False, shell=None, istream=None)
Popen(['git', 'version'], cwd=/app, universal_newlines=False, shell=None, istream=None)
patching TagDescriptor
Operations to perform:
  Apply all migrations: admin, auditlog, auth, authtoken, contenttypes, custom_field, defectDojo_engagement_survey, django_celery_results, dojo, sessions, sites, social_django, tagging, tastypie, watson
Running migrations:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/utils.py", line 82, in _execute
    return self.cursor.execute(sql)
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/mysql/base.py", line 71, in execute
    return self.cursor.execute(query, args)
  File "/usr/local/lib/python3.5/site-packages/MySQLdb/cursors.py", line 209, in execute
    res = self._query(query)
  File "/usr/local/lib/python3.5/site-packages/MySQLdb/cursors.py", line 315, in _query
    db.query(q)
  File "/usr/local/lib/python3.5/site-packages/MySQLdb/connections.py", line 239, in query
    _mysql.connection.query(self, query)
MySQLdb._exceptions.OperationalError: (1050, "Table 'watson_searchentry' already exists")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/commands/migrate.py", line 234, in handle
    fake_initial=fake_initial,
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/executor.py", line 245, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/migration.py", line 124, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/operations/models.py", line 92, in database_forwards
    schema_editor.create_model(model)
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/base/schema.py", line 307, in create_model
    self.execute(sql, params or None)
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/base/schema.py", line 137, in execute
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.5/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/utils.py", line 82, in _execute
    return self.cursor.execute(sql)
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/mysql/base.py", line 71, in execute
    return self.cursor.execute(query, args)
  File "/usr/local/lib/python3.5/site-packages/MySQLdb/cursors.py", line 209, in execute
    res = self._query(query)
  File "/usr/local/lib/python3.5/site-packages/MySQLdb/cursors.py", line 315, in _query
    db.query(q)
  File "/usr/local/lib/python3.5/site-packages/MySQLdb/connections.py", line 239, in query
    _mysql.connection.query(self, query)
django.db.utils.OperationalError: (1050, "Table 'watson_searchentry' already exists")
  Applying watson.0001_initial...Admin user: admin
enabling audit logging
Admin password: Initialization detected that the admin user admin already exists in your database.
If you don't remember the admin password, you can create a new superuser with:
$ docker-compose exec uwsgi /bin/bash -c 'python manage.py createsuperuser'
```

**Steps to reproduce**
1. Runs the initializer job against an empty database, it should run all the migrations successfully without exceptions
2. Performs an helm chart upgrade which executes again the initializer job. This time some expectations are thrown such as `"Table 'watson_searchentry' already exists"`, see above.

**Expected behavior**

The initializer job should run multiple times without exception when the DB is already initialized. 

**Deployment method** *(select with an `X`)*
- [x] Kubernetes
- [ ] Docker
- [ ] setup.bash / legacy-setup.bash

**Environment information**
 - Operating System: [e.g. Ubuntu 18.04]
 - DefectDojo Commit Message: [use `git show -s --format="%h: %s [%ci]"`]

**Sample scan files** (optional)
If applicable, add sample scan files to help reproduce your problem.

**Screenshots** (optional)

**Console logs** (optional)
If applicable, add console logs to help explain your problem.

**Additional context** (optional)

**Comment by valentijnscholten** [comment]
Does this happen always or only when the first run of the migration fails for the django-watson migrations that are not supported on InnoDB?

**Comment by ccojocar** [comment]
It happens always after the first run. It might be related to django-watson migrations not supporting InnoDB, but the table seems to be created by the first run. Maybe it is incomplete, otherwise the migration should detect that the table exist and shouldn't try to run again.

**Comment by madchap** [comment]
FWIW, the initializer variable (`DD_INITIALIZE`) in my case is set to False in Production, so it runs only once -- and so it also does not wipe out System Settings if you've already done some setup.

**Comment by valentijnscholten** [comment]
These migrations are generated by `django-watson` and run by `django-migrations`. Migrations can be atomic on sqlite and postgres but not on MySQL. The error here comes from part of the migration succeeding and then running into an error (because it is trying to create an myisam table which isn't supported). So it fails and the migration is only partly applied. On the next run the whole migration file is tried again, but it fails on the first step because that step was already completed succesfully. Nothing we can do about this. Maybe `django-watson` could split out the migration of the myisam tables into seperated migrations. But in general developers are encouraged to squash mirrations, so splitting would go against that.

![image](https://user-images.githubusercontent.com/4426050/87580385-ba0e7780-c6d7-11ea-90cd-2a9593a3cf7e.png)
